### PR TITLE
Add initialize method params 'capabilities' and 'initializationOptions'

### PR DESCRIPTION
**Pull request by dan-t**

### CODE REVIEW

File: autoload/lsp.vim [hunk 1]
```diff
@@ -298,15 +298,28 @@ function! s:ensure_init(buf, server_name, cb) abort
         return
     endif
 
+    if has_key(l:server_info, 'capabilities')
+        let l:capabilities = l:server_info['capabilities']
+    else
+        let l:capabilities = {}
+    endif
+
+    if has_key(l:server_info, 'initialization_options')
```
Comment (prabirshrestha) [inline autoload/lsp.vim]: can you give an example of the server that uses this?

File: autoload/lsp.vim [hunk 2]
```diff
@@ -298,15 +298,28 @@ function! s:ensure_init(buf, server_name, cb) abort
         return
     endif
 
+    if has_key(l:server_info, 'capabilities')
+        let l:capabilities = l:server_info['capabilities']
```
Comment (prabirshrestha) [inline autoload/lsp.vim]: when would you want to manually set this? Is this when you want to disable certain feature of the lang server? Just curious.

File: autoload/lsp.vim (hunk 1 above)
Comment (dan-t) [reply autoload/lsp.vim]: [cquery](https://github.com/jacobdufault/cquery) uses at least `initializationOptions`, but both are part of the langugae server protocol for [initialize](https://github.com/Microsoft/language-server-protocol/blob/master/protocol.md#initialize).

Comment (prabirshrestha) [reply autoload/lsp.vim]: Great. Could you add a wiki page on how to use cquery at https://github.com/prabirshrestha/vim-lsp/wiki/Servers

**Comment by crapp** [comment]
Whats keeping you from merging this pull request @prabirshrestha. There already is a wiki page for cquery. Although I wish it would more specific :)

**Comment by prabirshrestha** [comment]
@crapp Was out of the country on holidays and I'm on-call so been busy. Should be able to merge by next week. Sorry for the wait.

**Comment by prabirshrestha** [comment]
Thanks. Merged

**Pull request merged**